src/accelerate/utils/launch.py

```diff
@@ -128,7 +128,10 @@ def prepare_multi_gpu_env(args: argparse.Namespace) -> Dict[str, str]:
     if main_process_port is None:
         main_process_port = 29500
 
-    if is_port_in_use(main_process_port):
+    # only need to check port availability in main process, in case we have to start multiple launchers on the same machine
+    # for some reasons like splitting log files.
+    need_port_check = num_machines <= 1 or int(args.machine_rank) == 0
```

Comment:
I don't think the check `num_machines <= 1` is needed, wdyt?

Reply:
Cause we don't have a default value for machine_rank and if num_machines <=1, it is highly possible that machine_rank is None.

Reply:
Oh, okay, just checked again and because the `machine_rank` defaults to 0, we need the check on `num_machines`, got it.